fsm.go

```diff
@@ -377,6 +390,18 @@ func (t transitionerStruct) transition(f *FSM) error {
 	return nil
 }
 
+// CancelTransition cancels an asynchrounous state change.
```

Comment:
This comment is great! But it will never be visible as this is a private method on a private struct. Can you move it to the public `CancelTransition()` method? If you can also update the public `Transition()` comment to match this quality it would be a welcome bonus!